sql/insert.go

```diff
@@ -186,3 +224,94 @@ func (p *planner) processColumns(tableDesc *TableDescriptor,
 
 	return cols, nil
 }
+
+func (p *planner) makeDefaultExprs(cols []ColumnDescriptor) ([]parser.Expr, error) {
+	// Check to see if any of the columns have DEFAULT expressions. If there are
+	// no DEFAULT expressions, we don't bother with constructing the defaults map
+	// as the defaults are all NULL.
+	haveDefaults := false
+	for _, col := range cols {
+		if col.DefaultExpr != nil {
+			haveDefaults = true
+			break
+		}
+	}
+	if !haveDefaults {
+		return nil, nil
+	}
+
+	// Construct a SELECT statement with an output for each column. The output is
+	// either the DEFAULT expression or NULL if no default expression was
+	// configured.
+	var buf bytes.Buffer
+	buf.WriteString("SELECT ")
+	for i, col := range cols {
+		if i > 0 {
+			buf.WriteString(", ")
+		}
+		if col.DefaultExpr != nil {
+			buf.WriteString(*col.DefaultExpr)
+		} else {
+			buf.WriteString("NULL")
+		}
+	}
+
+	sql := buf.String()
+	stmts, err := parser.Parse(sql, parser.Syntax(p.session.Syntax))
```

Comment:
I don't see an easy way to build the AST directly. What would I store in `ColumnDescriptor` in order to build the AST for `now()`? Or, more interestingly, `extract(millisecond from now())`? While the expression is variable free, it can be arbitrarily complex.

You're correct about the use of `p.session.Syntax`. We actually only use the syntax setting for scanning tokens and always output "traditional" syntax. So this should probably be `parser.ParseTraditional`.